gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.base;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.nessie.NessieCatalog;
+import org.apache.iceberg.spark.SparkSchemaUtil;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.types.StructType;
+import org.projectnessie.model.Content;
+import scala.Tuple2;
+
+/** DDL + DML functionality for the "IdentifiedResult" table. */
+public final class IdentifiedResultsRepo {
+
+  private static final String COL_TYPE = "type";
+  private static final String COL_GC_RUN_ID = "gcRunId";
+  private static final String COL_GC_RUN_START = "gcRunStart";
+  private static final String COL_CONTENT_ID = "contentId";
+  private static final String COL_EXPIRED_CONTENTS = "expiredContents";
+  private static final String COL_EXPIRED_AT_REFERENCE_NAME = "expiredAtReferenceName";
+
+  private static final String TYPE_CONTENT = "GC_CONTENT";
+  // marker row indicates that identify task is completed for that run id
+  // and the results for that run id is consumable.
+  private static final String TYPE_GC_MARKER = "GC_MARK";
+
+  private static final String SCHEMA_DDL =
```

Comment:
I think calling create from the iceberg apis or from the dataframe api would be easier. We can define the schema as avro instead of as a long and hard to read string